tests/unit/features/RandomFourierDotFeatures_unittest.cc

```diff
@@ -176,3 +176,30 @@ TEST(RandomFourierDotFeatures, add_to_dense_test)
 
 }
 
+TEST(RandomFourierDotFeatures, random_coeff_dim)
+{
+	int32_t num_dims = 50;
+	int32_t vecs = 5;
+	int32_t D = 100;
+
+	SGMatrix<int32_t> data(num_dims, vecs);
+	for (index_t i=0; i<vecs; i++)
+	{
+		for (index_t j=0; j<num_dims; j++)
+			data(j,i) = i+j;
+	}
+
+	auto d_feats = std::make_shared<DenseFeatures<int32_t>>(data);
+	SGVector<float64_t> params(1);
```

Comment:
this is another thing to refactor, this vector of params (in another PR) rather I would take a kernel instance and get the parameters from that

Reply:
Yeah, that's apt